metricbeat/module/vsphere/datastorecluster/_meta/fields.yml

```
@@ -4,6 +4,10 @@
   description: >
     Datastore Cluster
   fields:
+    - name: id
+      type: keyword
+      description: >
+        The Datastore Cluster ID.
```

Minor nit to keep consistency

```suggestion
        Datastore Cluster ID.
```